### PR TITLE
Updates to aid compiling parent selectors and lists

### PR DESCRIPTION
**Pull request by niaccurshi**
references #310, #558 and #556

### CODE REVIEW

File: src/Parser.php [hunk 1]
```diff
@@ -1211,24 +1211,34 @@ protected function expression(&$out)
     {
         $s = $this->seek();
 
+        if ($this->literal('&',false)) {
```
Comment (robocoder) [inline src/Parser.php]: If you get a chance, please update your PR for PSR2.

In addition, our coding convention would have:
* a space after the comma for function arguments,
* a blank line before `return` statements

File: src/Parser.php [hunk 2]
```diff
@@ -1302,21 +1350,6 @@ protected function value(&$out)
     {
         $s = $this->seek();
 
-        if ($this->literal('url(') && $this->match('data:([a-z]+)\/([a-z0-9.+-]+);base64,', $m, false)) {
-            $len = strspn($this->buffer, 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwyxz0123456789+/=', $this->count);
```
Comment (robocoder) [inline src/Parser.php]: Why are you removing this optimization?

**Comment by niaccurshi** [comment]
As per the other pull request, this is clearly on top of an out of date branch somehow, and I'll review and resolve into a different request.

**Comment by robocoder** [comment]
Ok. If you can, please also add to your testing:
```
$list: [black];
$list: join($list, white, comma);

p {
  background: linear-gradient($list);
}
```

**Pull request closed without merging**